src/fosslight_dependency/_package_manager.py

```
@@ -436,33 +446,49 @@ def json = groovy.json.JsonOutput.prettyPrint(groovy.json.JsonOutput.toJson(resu
+
     def add_allDeps_in_gradle(self, gradle_file):
         config = android_config if self.package_manager_name == const.ANDROID else gradle_config
         is_kts = gradle_file == 'build.gradle.kts'
+
+        try:
+            with open(gradle_file, 'r', encoding='utf-8') as f:
+                data = f.read()
+            if 'task allDeps' in data or 'tasks.register("allDeps"' in data or 'tasks.register(\'allDeps\'' in data:
+                logger.info('allDeps task already exists in build.gradle. Skip injection and execution.')
+                return False
+        except Exception as e:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)